espresso/core/java/androidx/test/espresso/ViewHierarchyRenderer.java

```diff
@@ -0,0 +1,24 @@
+package androidx.test.espresso;
+
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Renders a view hierarchy into a human readable string.
+ */
+public interface ViewHierarchyRenderer {
+
+  String render(@NonNull View rootView, @Nullable ProblemViews problemViews);
+
+  class ProblemViews {
+    @NonNull public final String suffix;
+    @NonNull public final List<View> views;
```

Comment:
Packed these two params into one wrapper type as it's an all or nothing kind of situation.

Reply:
The definition of "Problem" views seem pretty vague. If this is a view hierarchy renderer, why does it care about whether a view is "problematic" or not? It appears to me that the only responsibility of this interface might be rendering the hierarchy of a view.

Reply:
Okay, I see where you are coming from. In the HumanReadables, `getViewHierarchyErrorMessage` method gives the view hierarchy info and the error messages. However, I believe we might want to leave the notion of "problematic" views out of the scope of this view render.

Reply:
@adazh if it's "markedViews" would it make more sense?

Reply:
I would leave the nature of the views out and have this interface simply for the purpose of view hierarchy rendering.